pytorch_lightning/callbacks/quantization.py

```diff
@@ -174,7 +182,15 @@ def _check_feasible_fuse(self, model):
 
     def on_fit_start(self, trainer, pl_module):
         # QuantStub converts tensors from floating point to quantized
-        pl_module.quant = torch.quantization.QuantStub()
+        if self.max_num_inputs is None:
+            if pl_module.example_input_array is None:
+                raise MisconfigurationException(
+                    '`max_num_inputs` or `model.example_input_array must be provided to determine number of QuantStubs'
+                )
+            num_inps = len(pl_module.example_input_array)
+        else:
+            num_inps = self.max_num_inputs
+        pl_module.quants = nn.ModuleList([torch.quantization.QuantStub() for _ in range(num_inps)])
         # DeQuantStub converts tensors from quantized to floating point
         pl_module.dequant = torch.quantization.DeQuantStub()
```

Comment:
noob question: do we not need a list for these?

Reply:
`nn.ModuleList` is preferable to plain list since it makes PyTorch aware that the `QuantStubs` should be including in `self.parameters()`

Reply:
I was referring to

`pl_module.dequant = torch.quantization.DeQuantStub()`

Reply:
Ah I think that's fine without a list. The forum [post](https://discuss.pytorch.org/t/is-it-possible-to-quantize-model-have-more-than-one-input/115018/2) I'm going off showed that you only need unique `QuantStub` elements, but the `DeQuantStub` can be shared globally by all inputs